lib/mongoidable/document_extensions.rb

```diff
@@ -6,10 +6,71 @@ module DocumentExtensions
     extend ActiveSupport::Concern
 
     included do
-      ability_class = Mongoidable.configuration.ability_class.constantize
+      ability_class = Mongoidable.configuration.ability_class
       raise TypeError unless ability_class == Mongoidable::Ability || ability_class.superclass == Mongoidable::Ability
 
-      embeds_many :instance_abilities, class_name: Mongoidable.configuration.ability_class
+      embeds_many :instance_abilities, class_name: Mongoidable.configuration.ability_class.to_s do
+        def update(**attributes)
+          return unless should_update?(attributes)
+
+          attributes = ActionController::Parameters.new(attributes) if attributes.is_a?(Hash)
+          query = ability_query_attrs(attributes)
+
+          #
+          ability = instance_abilities.where(query).first
+          type = ability_type(attributes[:action])
+          method = parent_document.new_record? ? :build : :create
```

Comment:
Why bother doing a `:build` if you're going to do an `update_attributes` on the ability saving it and the parent record anyways?